### PR TITLE
[risk=low][RW-8978] Update more swaggers to OpenAPI 3

### PR DESCRIPTION
**Pull request by jmthibault79**
Update **rdr.yaml** to OpenAPI 3 by using https://editor.swagger.io/
Tested locally by running the `uploadReportingSnapshot` cron without errors.

Update **captcha.yaml** to OpenAPI 3 by using https://editor.swagger.io/
Tested locally by temporarily setting `enableCaptcha` to true and creating a new account.  It prompted me for captcha and accepted the response.

Update **mandrill.yaml** to OpenAPI 3 by using https://editor.swagger.io/
Tested locally by creating a new account, which sends an email.  I received it and the information was correct.

---
**PR checklist**

- [x] I have included an issue ID or "no ticket" in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [x] I have included a risk tag of the form `[risk=no|low|moderate|severe]` in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [x] I have manually tested this change and my testing process is described above.
- [ ] This change includes appropriate automated tests, and I have documented any behavior that cannot be tested with code.
- [ ] I have added explanatory comments where the logic is not obvious.
- One or more of the following is true:
  - [x] This change is intended to complete a JIRA story, so I have checked that all AC are met for that story.
  - [ ] This change fixes a bug, so I have ensured the steps to reproduce are in the Jira ticket or provided above.
  - [ ] This change impacts deployment safety (e.g. removing/altering APIs which are in use), so I have documented the impacts in the description.
  - [ ] This change includes a new feature flag, so I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later.
  - [ ] This change modifies the UI, so I have taken screenshots or recordings of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P).
  - [ ] This change modifies API behavior, so I have run the relevant E2E tests locally because API changes are not covered by our PR checks.
  - [ ] None of the above apply to this change.

### CODE REVIEW

File: api/src/main/java/org/pmiops/workbench/rdr/RdrMapper.java [hunk 1]
```diff
@@ -160,7 +160,7 @@ default RdrWorkspace.StatusEnum toModelStatus(WorkspaceActiveStatus workspaceAct
   @ValueMapping(source = "TRUE", target = "YES")
   @ValueMapping(source = "FALSE", target = "NO")
   @ValueMapping(source = "PREFER_NO_ANSWER", target = "PREFER_NOT_TO_ANSWER")
-  RdrDisability toRdrDisability(Disability d);
+  RdrYesNoPreferNot toRdrDisability(Disability d);
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/rdr/RdrMapper.java:163]: Not strictly necessary for this change, but the Disability entity field in the RDR yaml is exactly YesNoPreferNot, so we don't need it

File: api/src/main/resources/rdr.yaml [hunk 2]
```diff
@@ -1,694 +1,654 @@
...
   description: |
     Service for RDR Interactions.
   version: "0.1"
-#The host will get updated as per the environment it will be pmi-drc-api-test for dev and test
-host: "pmi-drc-api-test.appspot.com"
-schemes:
-  - "https"
-basePath: /
-
-securityDefinitions:
```
Comment (jmthibault79) [inline api/src/main/resources/rdr.yaml:17]: See line 645 for the new version `securitySchemes` - also I recommend turning off whitespace diffs!

File: api/src/main/resources/rdr.yaml [hunk 3]
```diff
@@ -1,694 +1,654 @@
...
-    enum:
-      - NO_EDUCATION
-      - GRADES_1_12
-      - COLLEGE_GRADUATE
-      - UNDERGRADUATE
-      - MASTER
-      - DOCTORATE
-      - PREFER_NOT_TO_ANSWER
-
-  Disability:
```
Comment (jmthibault79) [inline api/src/main/resources/rdr.yaml:417]: Not actually needed

File: api/src/test/java/org/pmiops/workbench/captcha/CaptchaVerificationServiceTest.java [hunk 4]
```diff
@@ -49,10 +49,10 @@ public void setUp() throws IOException, ApiException {
     config = WorkbenchConfig.createEmptyConfig();
     config.admin.loginUrl = "hostname";
     Mockito.when(cloudStorageClient.getCaptchaServerKey()).thenReturn("key");
-    mockCaptchaResponse("hostName", true);
```
Comment (jmthibault79) [inline api/src/test/java/org/pmiops/workbench/captcha/CaptchaVerificationServiceTest.java:52]: not really related, but while I'm here: this was always true

File: api/src/main/resources/rdr.yaml [hunk 5]
```diff
@@ -1,694 +1,654 @@
...
-      sexAtBirthOtherText:
-        type: string
-        description: None of these fully describe me, and I want to specify.  Max length 200.
-      yearOfBirth:
-        type: integer
-      yearOfBirthPreferNot:
-        type: boolean
-      disabilityHearing:
-        "$ref": "#/definitions/YesNoPreferNot"
-        description: Are you deaf or do you have serious difficulty hearing?
```
[the remark this answers is not in the document]
Comment (evrii) [reply api/src/main/resources/rdr.yaml:503]: Do you mind clarifying what you mean here? Are descriptions not allowed where there is a $ref field?

**Review by evrii — APPROVED** [review]
The changes look good to me. I ran the uploadReportingSnapshot cron without an issue. I also created a new user with captcha challenge and received email afterwards. All appears good.

**Comment by jmthibault79** [comment]
2 e2e failures are due to runtime creation - not related to this change.

**Comment by jmthibault79** [comment]
@evrii right, no descriptions when the field is a ref

**Pull request merged**